core/src/main/java/org/p2p/core/model/CurrencyMode.kt

```diff
@@ -0,0 +1,18 @@
+package org.p2p.core.model
+
+import org.p2p.core.utils.Constants
+import org.p2p.core.utils.Constants.FIAT_FRACTION_LENGTH
+import org.p2p.core.token.Token as UserToken
+
+sealed interface CurrencyMode {
+    data class Token constructor(val symbol: String, val fractionLength: Int) : CurrencyMode {
+        constructor(token: UserToken) : this(token.tokenSymbol, token.decimals)
+    }
+
+    sealed class Fiat(val fiatSymbol: String, val fiatAbbreviation: String) : CurrencyMode {
```

Comment:
or maybe we need to create another model class like `Token.Active` with info and use it here, whatcha think?

Reply:
this lgtm